### PR TITLE
feat(server): support storytelling

### PR DESCRIPTION
**Pull request by yk-eukarya**

### CODE REVIEW

File: server/gql/storytelling.graphql [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+  id: ID!
+  propertyId: ID!
+  pluginId: ID!
+  extensionId: ID!
+  linkedDatasetId: ID
+  page: Page!
+  property: Property
+  plugin: Plugin
+  extension: PluginExtension
+#  linkedDataset: Dataset
```
Comment (rot1024) [inline server/gql/storytelling.graphql:34]:
```suggestion
	sceneId: ID!
	scene: Scene
```

File: server/gql/storytelling.graphql [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+  index: Int
+}
+
+input MoveStoryInput {
+  sceneId: ID!
+  storyId: ID!
+  index: Int!
+}
+
+input DeleteStoryInput {
```
Comment (rot1024) [inline server/gql/storytelling.graphql:58]:
```suggestion
input DeleteStoryInput {
  sceneId: ID!
```

File: server/gql/storytelling.graphql [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+  page: Page!
+  story: Story!
+}
+
+extend type Mutation {
+  createStorytelling(input: CreateStoryInput!): StoryPayload!
+  updateStorytelling(input: UpdateStoryInput!): StoryPayload!
+  deleteStorytelling(input: DeleteStoryInput!): DeleteStoryPayload!
+  publishStorytelling(input: PublishStoryInput!): StoryPayload!
+  moveStorytelling(input: MoveStoryInput!): MoveStoryPayload!
```
Comment (rot1024) [inline server/gql/storytelling.graphql:179]:
```suggestion
  createStory(input: CreateStoryInput!): StoryPayload!
  updateStory(input: UpdateStoryInput!): StoryPayload!
  deleteStory(input: DeleteStoryInput!): DeleteStoryPayload!
  publishStory(input: PublishStoryInput!): StoryPayload!
  moveStory(input: MoveStoryInput!): MoveStoryPayload!
```

File: server/gql/storytelling.graphql [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+
+  createPage(input: CreatePageInput!): CreatePagePayload!
+  updatePage(input: UpdatePageInput!): UpdatePagePayload!
+  removePage(input: DeletePageInput!): DeletePagePayload!
+  movePage(input: MovePageInput!): MovePagePayload!
```
Comment (rot1024) [inline server/gql/storytelling.graphql:184]:
```suggestion
  createStoryPage(input: CreateStoryPageInput!): CreateStoryPagePayload!
  updateStoryPage(input: UpdateStoryPageInput!): UpdateStoryPagePayload!
  removeStoryPage(input: DeleteStoryPageInput!): DeleteStoryPagePayload!
  moveStoryPage(input: MoveStoryPageInput!): MoveStoryPagePayload!
```

File: server/gql/storytelling.graphql [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+# duplicatePage(input: DuplicatePageInput!): DuplicatePagePayload!
+
+  createBlock(input: CreateBlockInput!): CreateBlockPayload!
+  moveBlock(input: MoveBlockInput!): MoveBlockPayload!
+  removeBlock(input: RemoveBlockInput!): RemoveBlockPayload!
```
Comment (rot1024) [inline server/gql/storytelling.graphql:189]:
```suggestion
  createStoryBlock(input: CreateStoryBlockInput!): CreateStoryBlockPayload!
  moveStoryBlock(input: MoveStoryBlockInput!): MoveStoryBlockPayload!
  removeStoryBlock(input: RemoveStoryBlockInput!): RemoveStoryBlockPayload!
```

File: server/gql/storytelling.graphql [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+  title: String!
+  blocks: [Block!]!
+  swipe: Boolean!
+  layers: [ID!]!
+  swipeLayers: [ID!]
+  property: Property
+  createdAt: DateTime!
+}
+
+type Block implements Node {
```
Comment (rot1024) [inline server/gql/storytelling.graphql:24]:
```suggestion
type StoryBlock implements Node {
```

File: server/gql/storytelling.graphql [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+  alias: String!
+  property: Property
+  pages: [Page!]!
+  publishmentStatus: PublishmentStatus!
+  createdAt: DateTime!
+  updatedAt: DateTime!
+  publishedAt: DateTime
+}
+
+type Page implements Node {
```
Comment (rot1024) [inline server/gql/storytelling.graphql:13]:
```suggestion
type StoryPage implements Node {
```

File: server/gql/storytelling.graphql [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+  id: ID!
+  title: String!
+  blocks: [Block!]!
+  swipe: Boolean!
+  layers: [ID!]!
+  swipeLayers: [ID!]
+  property: Property
+  createdAt: DateTime!
```
Comment (rot1024) [inline server/gql/storytelling.graphql:21]:
```suggestion
  createdAt: DateTime!
  sceneId: ID!
  scene: Scene
```

File: server/gql/storytelling.graphql [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+  publishedAt: DateTime
+}
+
+type Page implements Node {
+  id: ID!
+  title: String!
+  blocks: [Block!]!
+  swipe: Boolean!
+  layers: [ID!]!
+  swipeLayers: [ID!]
```
Comment (rot1024) [inline server/gql/storytelling.graphql:19]:
```suggestion
  swipabbleLayers: [ID!]
```

File: server/gql/storytelling.graphql [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+  createdAt: DateTime!
+  updatedAt: DateTime!
+  publishedAt: DateTime
+}
+
+type Page implements Node {
+  id: ID!
+  title: String!
+  blocks: [Block!]!
+  swipe: Boolean!
```
Comment (rot1024) [inline server/gql/storytelling.graphql:17]:
```suggestion
  swipabble: Boolean!
```

File: server/gql/storytelling.graphql [hunk 11]
```diff
@@ -0,0 +1,190 @@
+type Story implements Node {
+  id: ID!
+  title: String!
+  alias: String!
+  property: Property
+  pages: [Page!]!
+  publishmentStatus: PublishmentStatus!
+  createdAt: DateTime!
+  updatedAt: DateTime!
+  publishedAt: DateTime
```
Comment (rot1024) [inline server/gql/storytelling.graphql:10]:
```suggestion
  publishedAt: DateTime
  sceneId: ID!
  scene: Scene
```

File: server/gql/storytelling.graphql [hunk 12]
```diff
@@ -0,0 +1,190 @@
...
+  property: Property
+  createdAt: DateTime!
+}
+
+type Block implements Node {
+  id: ID!
+  propertyId: ID!
+  pluginId: ID!
+  extensionId: ID!
+  linkedDatasetId: ID
```
Comment (rot1024) [inline server/gql/storytelling.graphql:29]:
```suggestion
  linkedDatasetId: ID
  pageId: ID!
```

File: server/pkg/builtin/main.go [hunk 13]
```diff
@@ -24,6 +24,7 @@ var pluginManifest = manifest.MustParseSystemFromBytes(pluginManifestJSON, nil,
 var (
 	PropertySchemaIDVisualizerCesium = property.MustSchemaID("reearth/cesium")
 	PropertySchemaIDInfobox          = property.MustSchemaID("reearth/infobox")
+	PropertySchemaIDPage             = property.MustSchemaID("reearth/page")
```
Comment (rot1024) [inline server/pkg/builtin/main.go:27]:
```suggestion
	PropertySchemaIDStoryPage             = property.MustSchemaID("reearth/storypage")
```

File: server/pkg/storytelling/page.go [hunk 14]
```diff
@@ -0,0 +1,248 @@
...
+)
+
+type Page struct {
+	id          PageID
+	property    PropertyID
+	title       string
+	swipe       bool
+	layers      LayerIDList
+	swipeLayers LayerIDList
+	blocks      BlockList
```
Comment (rot1024) [inline server/pkg/storytelling/page.go:18]: storyId is required here, or a story has pages?

Comment (yk-eukarya) [reply server/pkg/storytelling/page.go:18]: yes the story has pages

**Review by pyshx — APPROVED** [review]

**Pull request merged**